### ISSUE TITLE
acc value less than val_acc

### ISSUE THREAD

**Issue by saig07** [naokishibuya/car-behavioral-cloning#7]
Hi there, I trained the model from scratch and just added in the accuracy metric, however I am getting acc value less than val_acc. This is the output:
20000/20000 [==============================] - 121s - loss: 0.0338 - acc: 0.3356 - val_loss: 0.0172 - val_acc: 0.8464
Epoch 2/10
20000/20000 [==============================] - 102s - loss: 0.0309 - acc: 0.3292 - val_loss: 0.0149 - val_acc: 0.8321
Epoch 3/10
20000/20000 [==============================] - 99s - loss: 0.0299 - acc: 0.3304 - val_loss: 0.0165 - val_acc: 0.8607
Epoch 4/10
20000/20000 [==============================] - 100s - loss: 0.0296 - acc: 0.3266 - val_loss: 0.0166 - val_acc: 0.8464
Epoch 5/10
20000/20000 [==============================] - 104s - loss: 0.0281 - acc: 0.3313 - val_loss: 0.0149 - val_acc: 0.8750
Epoch 6/10
20000/20000 [==============================] - 105s - loss: 0.0280 - acc: 0.3247 - val_loss: 0.0218 - val_acc: 0.8464
Epoch 7/10
20000/20000 [==============================] - 109s - loss: 0.0266 - acc: 0.3267 - val_loss: 0.0128 - val_acc: 0.8643
Epoch 8/10
20000/20000 [==============================] - 103s - loss: 0.0260 - acc: 0.3263 - val_loss: 0.0185 - val_acc: 0.8464
Epoch 9/10
20000/20000 [==============================] - 103s - loss: 0.0255 - acc: 0.3316 - val_loss: 0.0118 - val_acc: 0.8357
Epoch 10/10
20000/20000 [==============================] - 111s - loss: 0.0254 - acc: 0.3327 - val_loss: 0.0153 - val_acc: 0.8643

**Comment by naokishibuya** [comment]
Hi @saig07 
First of all, I'm not quite sure why you need 'accuracy' metric as this is not a classification problem.  Having said that, Keras has a way to add a custom 'accuracy' so we may not be talking about the same thing.  Can you show me your code how 'accuracy' was added?

**Comment by saig07** [comment]
I have just added in "metrics=['accuracy']" in the following line of code:
model.compile(loss='mean_squared_error', optimizer=Adam(lr=args.learning_rate), metrics=['accuracy'])

**Comment by naokishibuya** [comment]
Hi @saig07 
I believe that 'accuracy' metric in Keras is for classification models.  So, I really can't comment on the output when it is used for regression problem like this project.  If you want to use an accuracy metric with this project, you'll need to write a custom function that converts the steering angle (both prediction and label) to some distinct values like 'left', 'right' and 'straight' (say, -1, 1, 0) to do the comparison.  The below web page has some details.

https://keras.io/metrics/

**Comment by saig07** [comment]
Is there anything other method that you could recommend for evaluation? Any help is much appreciated and thank you for the feedback.

**Comment by naokishibuya** [comment]
Nothing comes to my mind other than the 'mean_squared_error' used in the project.  As for improving the model performance (i.e., driving), I find it most useful to record driving data for some difficult to handle scenarios and use that to train the model.

**Comment by saig07** [comment]
Thank you for the help
